invenio_rdm_records/services/services.py

```diff
@@ -342,19 +342,86 @@ def delete_secret_link(
 
         return True
 
-    def get_pid(
-        self, id_, identity,
-    ):
-        """Reserve doi."""
-        #  provider =
-        # find_the_provider_by_the_param_value
-        # (<provider param value>)
-        #  return provider.reserve(...)
+    def get_provider(self, scheme, client=None):
```

Comment:
Can we isolate all this about PIDs in a new dedicated service?

Reply:
See first long comment :)